### ISSUE TITLE
Feature request: Datepicker: extended Date resolution

### ISSUE THREAD

**Issue by sandrocorradi** [sbb-design-systems/sbb-angular#169]
Spec: https://digital.sbb.ch/de/webapps/components/datepicker

*DoD*
- Provide `BusinessDateAdapter`, which is an extension of `NativeDateAdapter` in the core/datetime module
- Provide a constant `SBB_BUSINESS_DATE_ADAPTER` in the core/datetime module, which is `{ provide: DateAdapter, useClass: BusinessDateAdapter }` and can be added to NgModule providers
- Document this feature in core/datetime markdown and datepicker markdown

**Is your feature request related to a problem? Please describe.**
For BizApps it is necessary that entering a date will be as efficient as possible. 

**Describe the solution you'd like**
Enter "200919" should be resolved as "20.09.2019"

**Comment by kyubisation** [comment]
This is not compliant with the specification: https://digital.sbb.ch/en/webapps/components/datepicker

@Frenggi What are your thoughts on this?

**Comment by Frenggi** [comment]
Hi @kyubisation 
You‘re right. For business applications this feature request could be an added value, but for now this is not specified like that. Let us discuss the technical possibility for this in detail. Afterwards we would adapt the specification.

**Comment by kyubisation** [comment]
From a technical perspective, I could add a separate implementation, which could be used on demand.
e.g.
```ts
import { BUSINESS_DATE_ADAPTER } from '@sbb-esta/angular-core/datetime';
```
@Frenggi Would that be acceptable to you?

However I need very through specification in regards to what should be parsed how, ideally with a range of examples.
Maybe @sandrocorradi can provide some examples?

**Comment by sandrocorradi** [comment]
Hi all,

these 3 examples would be great, our datepicker in KABENAS all of these 3.




[cid:image001.png@01D56FB8.BEEC9920][Ein Bild, das Screenshot enthält.  Automatisch generierte Beschreibung][cid:image003.png@01D56FB8.BEEC9920]
[Ein Bild, das Screenshot enthält.  Automatisch generierte Beschreibung]
Von: kyubisation <notifications@github.com>
Antworten an: SchweizerischeBundesbahnen/sbb-angular <reply@reply.github.com>
Datum: Donnerstag, 19. September 2019 um 17:39
An: SchweizerischeBundesbahnen/sbb-angular <sbb-angular@noreply.github.com>
Cc: "Corradi Sandro (IT-SWE-CC2-UX)" <sandro.corradi@sbb.ch>, Mention <mention@noreply.github.com>
Betreff: Re: [SchweizerischeBundesbahnen/sbb-angular] Feature request: Datepicker: extended Date resolution (#169)


From a technical perspective, I could add a separate implementation, which could be used on demand.
e.g.

import { BUSINESS_DATE_ADAPTER } from '@sbb-esta/angular-core/datetime';

@Frenggi<https://github.com/Frenggi> Would that be acceptable to you?

However I need very through specification in regards to what should be parsed how, ideally with a range of examples.
Maybe @sandrocorradi<https://github.com/sandrocorradi> can provide some examples?

**Comment by kyubisation** [comment]
@sandrocorradi The image did not get attached. Can you add it manually or type it out?

**Comment by sandrocorradi** [comment]
these 3 examples would be great, our datepicker in KABENAS works fine with all of these 3.
![image002](https://user-images.githubusercontent.com/55390734/65409494-25edbf80-dde8-11e9-90b4-f60e25b3405d.png)
![image003](https://user-images.githubusercontent.com/55390734/65409500-2b4b0a00-dde8-11e9-9333-9125e29a23fb.png)
![image001](https://user-images.githubusercontent.com/55390734/65409517-33a34500-dde8-11e9-8567-1d1ed8041568.png)

**Comment by Frenggi** [comment]
Hi @kyubisation 
As discussed, I've adjusted the rules in the pattern datepicker so that the different format resolutions are more clearly defined. See https://digital.sbb.ch/en/webapps/components/datepicker

**Comment by kyubisation** [comment]
Closed via #360
